feat: cron STAC validate TDE-1136 (#596)
linz/topo-workflows: workflows/cron/README.md

# Contents:

> - [cron-stac-validata](#cron-stac-validate)

Reviewer: ```suggestion
- [cron-stac-validate](#cron-stac-validate)
```

- [cron-stac-validate-all-data](#cron-stac-validate-all-data)

# STAC validation

The goal of the following [Cron Workflows](https://argo-workflows.readthedocs.io/en/stable/cron-workflows/) is to check the validity of the STAC metadata published in the AWS Open Data Registries [NZ Elevation](https://registry.opendata.aws/nz-elevation/) and [NZ Imagery](https://registry.opendata.aws/nz-imagery/).

> > **_NOTE:_** To simplify the overall workflow deployment process, these `CronWorkflow` have one main task per registry. It looks like a duplication that could be avoid but as we are not using [`argo` CLI](https://argo-workflows.readthedocs.io/en/stable/walk-through/argo-cli/) to deploy the workflows - which allows parameter passing - we could not deploy one `CronWorkflow` per `uri` (or registry).

Reviewer: ```suggestion
> **_NOTE:_** To simplify the overall workflow deployment process, these `CronWorkflow`s have one main task per registry. It looks like a duplication that could be avoided but as we are not using [`argo` CLI](https://argo-workflows.readthedocs.io/en/stable/walk-through/argo-cli/) to deploy the workflows - which allows parameter passing - we could not deploy one `CronWorkflow` per `uri` (or registry).
```

> ## cron-stac-validate

Reviewer: Out of context it sounds like this validates everything. What would be a good name for it so that it's obvious that it only validates STAC metadata?

Author: `stac-validate` sounds like validating the STAC to me. Would you like to have a suffix `-metadata` as well?

Reviewer: Basically, it would be good to have two names for these jobs where it's obvious that one is a subset of the other. So something like

- stac-validate-metadata and stac-validate-metadata-and-checksums or
- stac-validate-only-metadata and stac-validate-all.

Reviewer: The new names still have the same problem - it's difficult to tell whether "cron-stac-validate" does more or less than "cron-stac-validate-checksums".

> Workflow that validates the STAC metadata by calling the [`stac-validate` argo-tasks command](https://github.com/linz/topo-workflows/blob/master/templates/argo-tasks/README.md#argo-tasksstac-validate) using the [`tpl-at-stac-validate`](https://github.com/linz/topo-workflows/blob/master/templates/argo-tasks/README.md#argo-tasksstac-validate). It does verify that the [STAC links](https://github.com/radiantearth/stac-spec/blob/master/collection-spec/collection-spec.md#link-object) are valid.

Reviewer: Shouldn't the `tpl-at-stac-validate` link point somewhere else?

Author: That was the "`stac-validate` argo-tasks command" that pointed to the wrong place!

> - schedule: **every day at 5am**

Reviewer: Which time zone?

Author: [NZ](https://github.com/linz/topo-workflows/blob/2efa5bfd468d517445c83e2d7c6787f36e2e3e18/workflows/cron/cron-stac-validate.yaml#L10)

Reviewer: Oh, they use non-standard time zone names. Welp.

## cron-stac-validate-all-data

> Workflow that validates the STAC metadata by calling the [`stac-validate` argo-tasks command](https://github.com/linz/topo-workflows/blob/master/templates/argo-tasks/README.md#argo-tasksstac-validate) using the [`stac-validate-parallel`](https://github.com/linz/topo-workflows/blob/master/workflows/stac/README.md#stac-validate-parallel)

Reviewer: Missing end of sentence.

> It also validate that the [STAC assets](https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#assets) are valid. Verifying all asset (TIFF files) checksums is expensive, so this workflow is ran less often than the [cron STAC validate](#cron-stac-validate).

Reviewer: Do you want to put an empty line above this, to treat it as a paragraph?

Reviewer: ```suggestion
It also validate that the [STAC assets](https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#assets) are valid. Verifying all asset (TIFF files) checksums is expensive, so this workflow is run less often than [cron STAC validate](#cron-stac-validate).
```
